### PR TITLE
docs: CI/CD update - branches / contract_requiring_verification_published webhook

### PR DESCRIPTION
**Pull request by YOU54F**
Update Pactflow CI/CD Workshop

- update to use contract_requiring_verification_published event/webhook
- update to use branches over tags
- rename workshop to Consumer-Driven Contract Testing (to contrast against Bi-Directional Contract Testing) workshops
- some other niceties (file locations, git commit commands)

### CODE REVIEW

**Comment by bethesque** [comment]
Hoorah for this update happening!

File: website/docs/docs/workshops/ci-cd-legacy/set-up-ci/04_configure_webhook.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+* When the provider changes (to make sure it does not break any existing consumer expectations)
+* When a pact changes (to see if the provider is compatible with the new expectations)
+
+To ensure that the verification step is run whenever a pact changes, we need to configure a Pactflow webhook to trigger a provider verification build in Github Actions. You can see the configuration for this build in `.github/workflows/verify_changed_pact.yml` in the provider project. Rather than verifying the pacts with the configured tags, this build just verifies the pact that has changed. This is achieved by passing the URL of the changed pact to the build via a parameter in the body of the webhook request.
+
+The Pactflow webhook will need a Github access token to be able to trigger the build in Github. We don't want the Github token to be stored in clear text in the webhook, so we will create a secret in Pactflow to contain token.
+
+1. Create a Github token.
+    1. In Github:
+        1. Open the `Personal acesss tokens page`
```
Comment (Ilia) [inline website/docs/docs/workshops/ci-cd-legacy/set-up-ci/04_configure_webhook.md:17]: @YOU54F while you are making changes, do you want to fix this typo as well? acesss->access? its in both legacy and new

**Comment by YOU54F** [comment]
Hey @Ilia 

I've addressed these

1. [Minor] Setup deployment environment - "Settings > Environments", its more a UX issue, but there was no "settings", there was a gear icon, maybe worth including that as an imagine in the docs
1. [Minor] Setup deployment environment - "production" environment, since I had a brand new setup, I already had a production environment, and thought for some reason it will allow me to create another, but it didnt and showed an error, at this stage I wasnt sure if I could re-use the one there, OR i should delete and create another
1. [Minor] Configure consumer pipeline - we could instruct the user to just update the files in Github Editor and commit, potentially saving the step of cloning locally and running npm (unsure though if this is required in the next steps)
1. [Minor] Configure consumer pipeline - step 2, once github picked up the commit, my build errored, i thought I did something wrong, but it actually should still error. I know the text indicates that it should, but maybe a warning box may be good, as this This build should now successfully publish the pact, but it will fail on the can-i-deploy step before it tries to deploy gets lost

Not done this one yet, but it's not a bad shout, having it as a secret means the user can just add it during the pb token step 👍🏾 

1. [Minor] Configure provider pipeline - There are TWO workflow files - we could simplify this with a shared env across workflows, there is some docs here https://arinco.com.au/blog/github-actions-share-environment-variables-across-workflows/ or maybe even have it as a secret, which is shared across all workflows - more a recommendation

File: website/docs/docs/workshops/ci-cd.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 ---
 id: ci-cd
-title: Pactflow CI/CD Workshop
+title: Consumer-Driven Contract Testing Workshop
```
[the remark this answers is not in the document]
Comment (YOU54F) [reply website/docs/docs/workshops/ci-cd.md]: gone for 

- `Consumer-Driven Contract Testing CI/CD Workshop`
- `Consumer-Driven Contract Testing CI/CD Workshop (Legacy)`

**Comment by bethesque** [comment]
> There are TWO workflow files - we could simplify this with a shared env across workflows, 

Sometimes duplicated+simple is better than shared+moreComplicated ;)

File: website/docs/docs/workshops/ci-cd/set-up-ci/04_configure_webhook.md [hunk 3]
```diff
@@ -8,49 +8,58 @@ When using Pact in a CI/CD pipeline, there are two reasons for a pact verificati
...
+To ensure that the verification step is run whenever a pact changes, we need to configure a Pactflow webhook to trigger a provider verification build in Github Actions.
+
+You can see the configuration for this build in `.github/workflows/contract_requiring_verification_published.yml` in the provider project.
+
+Rather than verifying the pacts with the configured [consumer version selectors](https://docs.pact.io/pact_broker/advanced_topics/consumer_version_selectors), it triggers once for each of the following provider versions that are missing a verification result for the newly published pact:
+
+* the latest version from the provider's main branch
+* any version currently deployed to an environment
+  
+This is achieved by passing the URL of the changed pact to the build via a parameter in the body of the webhook request, as well as the provider version number and the provider branch of the head, test and production versions.
```
Comment (bethesque) [inline website/docs/docs/workshops/ci-cd/set-up-ci/04_configure_webhook.md:20]: head -> main

File: website/docs/docs/workshops/ci-cd/set-up-ci/04_configure_webhook.md [hunk 4]
```diff
@@ -8,49 +8,58 @@ When using Pact in a CI/CD pipeline, there are two reasons for a pact verificati
...
-            * Description:
-
-                ```
-                Pact changed webhook for pactflow-example-provider
-                ```
+            * Description: `Contract requiring verification published webhook for pactflow-example-provider`
             * Consumer: leave as "ALL"
             * Provider: select `pactflow-example-provider`
-            * Events: select `Contract published with changed content or tags`
+            * Events: select `Contract published that requires verification`
```
Comment (bethesque) [inline website/docs/docs/workshops/ci-cd/set-up-ci/04_configure_webhook.md:56]: woot!

File: website/docs/docs/workshops/ci-cd/set-up-local-development/02_run_the_applications.md [hunk 5]
```diff
@@ -3,11 +3,20 @@ id: run-the-applications
...
+
+Provider API listening on port 8080...
+```
 
 Open a separate terminal for the consumer.
 
-Before starting the consumer, create a `.env` file in the root of the project and add:
+Before starting the consumer, we need to configure it so it knows the address of our running provider.
+
+create a `.env` file in the root of the project and add:
```
Comment (bethesque) [inline website/docs/docs/workshops/ci-cd/set-up-local-development/02_run_the_applications.md:19]: Create

File: website/docs/docs/workshops/ci-cd/workshop/06_verifying_feature_pacts.md [hunk 6]
```diff
@@ -5,65 +5,75 @@ title: Strategies for verifying feature pacts
...
+When enabled, it will cause the provider to automatically verify, in pending mode, any "outstanding" pacts (ie. ones for which it has not yet published a successful result), as well as the pacts that were explicitly specified in the consumer version selector list (such as a specified `branch`).
 
-A "work in progress" pact is a pact that is the latest for its tag that does not have any successful verification results (ie. is still pending) for the configured provider tag . At this stage in the exercise, the `feat/new-field` pact is still in pending state for the `master` branch of the provider, so it is considered a "work in progress" pact for master.
+A "work in progress" pact is a pact that is the latest for its branch that does not have any successful verification results (ie. is still pending) for the configured provider branch . At this stage in the exercise, the `feat/new-field` pact is still in pending state for the main branch of the provider, so it is considered a "work in progress" pact for `master`.
```
Comment (bethesque) [inline website/docs/docs/workshops/ci-cd/workshop/06_verifying_feature_pacts.md:38]: It's slightly more complicated than this now. My brain is not working so I'll come back to this.

**Review by bethesque — APPROVED** [review]

**Comment by bethesque** [comment]
This makes me so happy!!!! 🎉

File: website/docs/docs/workshops/ci-cd/workshop/06_verifying_feature_pacts.md (hunk 6 above)
Comment (bethesque) [reply website/docs/docs/workshops/ci-cd/workshop/06_verifying_feature_pacts.md:38]: Now I think about it, it's not worth confusing people with the full technical details of how WIP pacts are selected at this stage. What you've got is 👍🏽

**Comment by YOU54F** [comment]
Will merge it and go through with the SE team next week for some user feedback :)

**Pull request merged**